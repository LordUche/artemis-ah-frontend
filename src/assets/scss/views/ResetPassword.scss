@import "../base.scss";

.reset_body{
  padding: 0;
  text-align: center;
  margin: 10% 0;

  &_content {
    display: inline-block;
    padding: 20px 60px;
    border-radius: 5px;
    box-shadow: 2px 2px 4px 1px rgba(0, 0, 0, 0.2);
  }

  &_reset_header {
    margin: 30px 0;
    font-size: 2rem;
    font-weight: $bold-font;
  }

  &_reset_input, &_btn_save {
    width: 30rem;
    margin-top: 20px;
  }

  &_response_text {
    margin-top: 15px;
    font-size: 15px;
    text-transform: capitalize;
  }

  &_btn_save {
    background: $primary-color;
    color: $white-color;
    margin: 30px 0;
    outline: none;
  }
}

@media screen and (max-width: 500px){
  .reset_body{
    margin: 10% 0;

    &_content {
      padding: 22px;
    }
  }
}