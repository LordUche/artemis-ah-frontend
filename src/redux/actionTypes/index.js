export const CHANGE_NAME = 'CHANGE_NAME';

// Get all articles Action types
export const GET_ARTICLES = 'GET_ARTICLES';
export const GET_ARTICLES_ERROR = 'GET_ARTICLES_ERROR';
export const ARTICLES_LOADING = 'ARTICLES_LOADING';

// Auth Action types
export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const CLEAR_AUTH_ERROR = 'CLEAR_AUTH_ERROR';
export const AUTH_LOADING = 'AUTH_LOADING';

// Profile action types
export const PROFILE_USER_DETAILS_FETCHED = 'PROFILE_USER_DETAILS_FETCHED';
export const PROFILE_USER_DETAILS_FETCH_ERROR = 'PROFILE_USER_DETAILS_FETCH_ERROR';

export const PROFILE_ARTICLES_FETCHING = 'PROFILE_ARTICLES_FETCHING';
export const PROFILE_ARTICLES_FETCHED = 'PROFILE_ARTICLES_FETCHED';
export const PROFILE_ARTICLES_FETCH_ERROR = 'PROFILE_ARTICLES_FETCH_ERROR';

export const PROFILE_FOLLOWERS_FETCHING = 'PROFILE_FOLLOWERS_FETCHING';
export const PROFILE_FOLLOWERS_FETCHED = 'PROFILE_FOLLOWERS_FETCHED';
export const PROFILE_FOLLOWERS_FETCH_ERROR = 'PROFILE_FOLLOWERS_FETCH_ERROR';

export const PROFILE_FOLLOWING_FETCHING = 'PROFILE_FOLLOWING_FETCHING';
export const PROFILE_FOLLOWING_FETCHED = 'PROFILE_FOLLOWING_FETCHED';
export const PROFILE_FOLLOWING_FETCH_ERROR = 'PROFILE_FOLLOWING_FETCH_ERROR';

export const PROFILE_DETAILS_UPDATING = 'PROFILE_DETAILS_UPDATING';
export const PROFILE_DETAILS_UPDATED = 'PROFILE_DETAILS_UPDATED';
export const PROFILE_DETAILS_UPDATE_ERROR = 'PROFILE_DETAILS_UPDATE_ERROR';

export const PROFILE_RESET_EDIT_STATE = 'PROFILE_RESET_EDIT_STATE';

export const PROFILE_RESET = 'PROFILE_RESET';
// Articles Action types
export const FETCH_TAGS = 'FETCH_TAGS';
export const FETCH_TAGS_ERROR = 'FETCH_TAGS_ERROR';
export const CREATE_ARTICLE = 'CREATE_ARTICLE';
export const CREATE_ARTICLE_ERROR = 'CREATE_ARTICLE_ERROR';
export const CLEAR_ARTICLE_ERROR = 'CLEAR_ARTICLE_ERROR';
export const PUBLISHING_ARTICLE = 'PUBLISHING_ARTICLE';

// Sign Up Action types
export const SIGN_UP = 'SIGN_UP';
export const SIGN_UP_ERROR = 'SIGN_UP_ERROR';

export const EDIT_ARTICLE = 'EDIT_ARTICLE';
export const SAVE_EDITED_ARTICLE = 'SAVE_EDITED_ARTICLE';

export const OPEN_DELETE_CONFIRMATION_MODAL = 'OPEN_DELETE_CONFIRMATION_MODAL';
export const CLOSE_DELETE_CONFIRMATION_MODAL = 'CLOSE_DELETE_CONFIRMATION_MODAL';
export const DELETE_ARTICLE = 'DELETE_ARTICLE';

export const FETCH_DELETE_ERROR = 'FETCH_DELETE_ERROR';

// Record Action types
export const GETTING_ARTICLE = 'GETTINg_ARTICLE';
export const GOT_ARTICLE = 'GOT_ARTICLE';
export const ERROR_GETTING_ARTICLE = 'ERROR_GETTING_ARTICLE';

export const NEW_NOTIFICATION = 'NEW_NOTIFICATION';
export const GET_NOTIFICATIONS = 'GET_NOTIFICATIONS';
export const READ_NOTIFICATION = 'READ_NOTIFICATION';
