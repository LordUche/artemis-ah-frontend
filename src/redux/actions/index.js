// Base url
const BASE_URL = 'https://authorshaven.herokuapp.com/api';

export default BASE_URL;
