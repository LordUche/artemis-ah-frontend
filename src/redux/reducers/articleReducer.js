import {
  CREATE_ARTICLE,
  CREATE_ARTICLE_ERROR,
  CLEAR_ARTICLE_ERROR,
  PUBLISHING_ARTICLE,
  SAVE_EDITED_ARTICLE,
  EDIT_ARTICLE,
  OPEN_DELETE_CONFIRMATION_MODAL,
  CLOSE_DELETE_CONFIRMATION_MODAL,
  GET_ARTICLES,
  GET_ARTICLES_ERROR,
  GETTING_ARTICLE,
  GOT_ARTICLE,
  ERROR_GETTING_ARTICLE
} from '../actionTypes';

export const initialState = {
  articleData: {},
  articles: [],
  totalNumberOfArticles: 0,
  limit: 0,
  loading: true,
  errors: {},
  isPublishing: false,
  articleCardData: JSON.parse(localStorage.getItem('cardData')) || {},
  confirmationModal: false,
  updatedArticle: {},
  isGetting: false,
  articleGotten: {},
  newArticleSlug: ''
};

/**
 * @param {object} state - The initial state
 * @param {object} {*} - destructured type object
 * @returns {object} - The transformed state
 */
const articleReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case GET_ARTICLES:
      return {
        ...state,
        articles: payload.articles,
        totalNumberOfArticles: payload.total,
        limit: payload.limit,
        loading: false
      };
    case GET_ARTICLES_ERROR:
      return {
        ...state,
        errors: payload,
        loading: false
      };
    case CREATE_ARTICLE:
      return {
        ...state,
        isPublishing: false,
        newArticleSlug: payload.article.slug,
        articleData: payload.article
      };
    case CREATE_ARTICLE_ERROR:
      return {
        ...state,
        isPublishing: false,
        errors: payload
      };
    case CLEAR_ARTICLE_ERROR:
      return {
        ...state,
        isPublishing: false,
        isGetting: false,
        newArticleSlug: '',
        errors: {}
      };
    case PUBLISHING_ARTICLE:
      return {
        ...state,
        isPublishing: true
      };
    case EDIT_ARTICLE:
      return {
        ...state,
        articleCardData: payload
      };
    case SAVE_EDITED_ARTICLE:
      return {
        ...state,
        isPublishing: false,
        updatedArticle: payload
      };
    case OPEN_DELETE_CONFIRMATION_MODAL:
      return {
        ...state,
        confirmationModal: true
      };
    case CLOSE_DELETE_CONFIRMATION_MODAL:
      return {
        ...state,
        confirmationModal: false
      };
    case GETTING_ARTICLE:
      return {
        ...state,
        isGetting: true,
        errors: {}
      };
    // eslint-disable-next-line no-case-declarations
    case GOT_ARTICLE:
      const { article, clap } = payload;
      return {
        ...state,
        isGetting: false,
        articleGotten: { ...article, clap },
        errors: {}
      };
    case ERROR_GETTING_ARTICLE:
      return {
        ...state,
        isGetting: false,
        articleDetailsGotten: {},
        errors: payload
      };
    default:
      return state;
  }
};

export default articleReducer;
