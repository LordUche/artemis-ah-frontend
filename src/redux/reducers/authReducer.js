import {
  LOGIN_USER, LOGIN_ERROR, AUTH_LOADING, CLEAR_AUTH_ERROR
} from '../actionTypes';

/**
 * @description function to return initial state
 * @param {object} localStorage the local storage to query (to be mocked for testing)
 * @returns {object} the initial state
 */
export const initialState = localStorage => ({
  isLoggedIn: !!localStorage.authorsHavenToken,
  token: localStorage.authorsHavenToken || '',
  errorMessages: {},
  loading: false
});

/**
 * @description reducer for auth operations
 * @param {object} state the current state
 * @param {object} action the action dispatched
 * @returns {object} the new state
 */
const authReducer = (state = initialState(localStorage), { type, payload }) => {
  switch (type) {
    case LOGIN_USER:
      return {
        ...state,
        token: payload.token,
        isLoggedIn: true,
        loading: false
      };
    case AUTH_LOADING:
      return {
        ...state,
        loading: true
      };
    case LOGIN_ERROR:
      return {
        ...state,
        errorMessages: payload.message ? { errors: { general: payload.message } } : payload,
        loading: false
      };
    case CLEAR_AUTH_ERROR:
      return {
        ...state,
        errorMessages: []
      };
    default:
      return state;
  }
};

export default authReducer;
