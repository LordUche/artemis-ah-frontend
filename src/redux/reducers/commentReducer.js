import {
  GET_COMMENTS,
  POST_COMMENT,
  POST_COMMENT_ERROR,
  COMMENT_LOADING,
  CLEAR_POSTED,
  LIKE_COMMENT_ERROR,
  TOGGLE_COMMENT_LIKE,
} from '../actionTypes';


export const initialState = {
  articleComments: [],
  errors: {},
  posted: false,
  loading: false
};

/**
 * @param {*} state
 * @returns {object} comment reducer
 */
export const commentReducer = (state = initialState, { type, payload }) => {
  let chosenComment, remainingComments;
  if (type === TOGGLE_COMMENT_LIKE || type === LIKE_COMMENT_ERROR) {
    const { articleComments } = state;
    const { id } = payload;
    chosenComment = articleComments.find(c => c.id === id);
    remainingComments = articleComments.filter(c => c.id !== id);
  }
  switch (type) {
    case GET_COMMENTS:
      return {
        ...state,
        articleComments: payload,
        errors: {}
      };
    case POST_COMMENT:
      return {
        ...state,
        posted: true,
        loading: false,
        errors: {}
      };
    case POST_COMMENT_ERROR:
      return {
        ...state,
        errors: payload.errors,
        loading: false
      };
    case COMMENT_LOADING:
      return {
        ...state,
        loading: true,
        errors: {}
      };
    case CLEAR_POSTED:
      return {
        ...state,
        posted: false,
        errors: {}
      };
    case TOGGLE_COMMENT_LIKE:
      chosenComment.hasLiked = !chosenComment.hasLiked;
      chosenComment.totalLikes = chosenComment.hasLiked
        ? chosenComment.totalLikes + 1 : chosenComment.totalLikes - 1;
      return {
        ...state,
        articleComments: [...remainingComments, chosenComment].sort((a, b) => b.id - a.id)
      };
    case LIKE_COMMENT_ERROR:
      chosenComment.hasLiked = !chosenComment.hasLiked;
      chosenComment.totalLikes = chosenComment.hasLiked
        ? chosenComment.totalLikes + 1 : chosenComment.totalLikes - 1;
      return {
        ...state,
        articleComments: [...remainingComments, chosenComment].sort((a, b) => b.id - a.id),
        errors: { message: 'Cannot perform action right now, please check your connection and try again later' }
      };
    default:
      return state;
  }
};

export default commentReducer;
