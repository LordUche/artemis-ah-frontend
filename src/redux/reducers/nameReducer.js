import CHANGE_NAME from '../actionTypes/index';

const inttialState = {
  name: 'adaeze'
};

/**
 * @description change name reducer
 * @param {*} state
 * @param {*} action
 * @returns {object} state
 */
export default function changeName(state = inttialState, action) {
  switch (action.type) {
    case CHANGE_NAME:
      return { ...state, name: action.payload };
    default:
      return state;
  }
}
