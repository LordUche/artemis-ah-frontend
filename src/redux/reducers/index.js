import { combineReducers } from 'redux';
import authReducer from './authReducer';
import userReducer from './userReducer';
import profileReducer from './profileReducer';
import tagReducer from './tagReducer';
import articleReducer from './articleReducer';
import notificationsReducer from './notificationsReducer';
import commentsReducer from './commentReducer';

const rootReducer = combineReducers({
  auth: authReducer,
  user: userReducer,
  profile: profileReducer,
  tags: tagReducer,
  article: articleReducer,
  notifications: notificationsReducer,
  comments: commentsReducer,
});

export default rootReducer;
