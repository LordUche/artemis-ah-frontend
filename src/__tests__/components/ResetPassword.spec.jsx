import 'babel-polyfill';
import React from 'react';
import moxios from 'moxios';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import ReduxPromise from 'redux-promise';
import ResetPassword from '../../views/ResetPasswordPage';
import reducers from '../../redux/reducers';

describe('ResetPassword Page Component', () => {
  const store = createStore(reducers, applyMiddleware(ReduxPromise));
  const wrapper = mount(
    <Provider store={store}>
      <Router>
        <ResetPassword />
      </Router>
    </Provider>
  );

  it('should match the given snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate form submit', () => {
    wrapper.find('form').simulate('submit');
    expect(wrapper.find('.reset_body_response_text')).toMatchObject({});
  });

  it('should simulate changes', () => {
    const newPassword = wrapper.find('[name="newPassword"]');
    const confirmPassword = wrapper.find('[name="confirmPassword"]');
    newPassword.simulate('change', { currentTarget: { value: 'Tyler' } });
    confirmPassword.simulate('change', { currentTarget: { value: 'John' } });
    const button = wrapper.find('[disabled="false"]');
    expect(button).toMatchObject({});
    expect(wrapper.find('button').text()).toEqual('Save');
  });
});

describe('should match the shallow snapshot', () => {
  const wrapper = shallow(<ResetPassword />);

  it('should match the state of component', () => {
    expect(wrapper.state('fetching')).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the state of component', () => {
    const resetPassword = new ResetPassword();
    const event = {
      preventDefault: () => 'prevent default'
    };

    resetPassword.handleSubmit(event);
  });
});
describe('Testing the delete article action', () => {
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
  });

  const resetPassword = new ResetPassword();
  const event = {
    preventDefault: () => 'prevent default'
  };

  resetPassword.handleSubmit(event);

  it('should change a password', async () => {
    const expectedResponse = 'password changed successfully';
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({ status: 200, response: expectedResponse });
    });

    await resetPassword.handleSubmit(event);
  });
});
