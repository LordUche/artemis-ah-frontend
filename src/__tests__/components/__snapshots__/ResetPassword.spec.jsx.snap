// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Page Component should match the given snapshot 1`] = `ReactWrapper {}`;

exports[`should match the shallow snapshot should match the state of component 1`] = `ShallowWrapper {}`;
